bundles/org.openhab.binding.helios/src/main/java/org/openhab/binding/helios/internal/handler/HeliosHandler221.java

```diff
@@ -169,27 +172,26 @@ public void initialize() {
                 logger.error("An exception occurred while initialising the SSL context : '{}'", e1.getMessage(), e1);
             }
 
-            heliosClient = ClientBuilder.newBuilder().sslContext(sslContext).hostnameVerifier(new HostnameVerifier() {
+            heliosClient = heliosClientBuilder.sslContext(sslContext).hostnameVerifier(new HostnameVerifier() {
                 @Override
                 public boolean verify(String hostname, javax.net.ssl.SSLSession sslSession) {
                     return true;
                 }
             }).build();
             heliosClient.register(new Authenticator(username, password));
 
-            baseTarget = heliosClient.target(BASE_URI);
+            baseTarget = heliosClient.target(BASE_URI.replace("{ip}", ipAddress));
             systemTarget = baseTarget.path(SYSTEM_PATH);
             logTarget = baseTarget.path(LOG_PATH);
             switchTarget = baseTarget.path(SWITCH_PATH);
 
             Response response = null;
             try {
-                response = systemTarget.resolveTemplate("ip", ipAddress).resolveTemplate("cmd", INFO)
-                        .request(MediaType.APPLICATION_JSON_TYPE).get();
-            } catch (NullPointerException e) {
+                response = systemTarget.resolveTemplate("cmd", INFO).request(MediaType.APPLICATION_JSON_TYPE).get();
+            } catch (NullPointerException | ProcessingException e) {
```

Comment:
Do you know why `NullPointerException` is caught here, can it still be thrown after your change?

Reply:
> Do you know why `NullPointerException` is caught here, can it still be thrown after your change?

No, I don't remember why I am catching that one - this is quite old code, there must have been a reason